### ISSUE TITLE
Add PATCHMEM?

### ISSUE THREAD

**Issue by TheMiningTeamYT** [JHRobotics/patcher9x#7]
A feature I’d like to see is the inclusion of the late Rloew’s PATCHMEM in this program.

**Comment by birdie-github** [comment]
https://archive.org/details/PATCHMEM

It's closed source. The author passed away. License unknown.

Not sure it it's legally OK to include it.

**Comment by TheMiningTeamYT** [comment]
On their website they said they are “…[making] it available to anyone who is interested in pursuing further endeavors” and also released the source code.
https://rloewelectronics.com/

**Comment by kristibektashi** [comment]
Yes I agree this should be added but as an optional feature that is triggered with an argument (such as patcher9x -moreram)

**Comment by Manawyrm** [comment]
> It's closed source.

The source code is available here:  
https://rloewelectronics.com/distribute/PATCHMEM/PRO7.2/
(broken SSL cert, but the site works)

**Comment by pappyN4** [comment]
I don't want to add to scope creep, but PATCH137 and the SATA patch from rloew are also interesting.  Both make changes to esdi_506.pdr

**Comment by Torinde** [comment]
Related patches from http://lonecrusader.x10host.com/rloew/rloew.html
- PATCH1TB: Patches Windows 9x to correctly handle Partitions larger than 1TiB.
- TBPLUS: Allows Windows 98SE to use hard drives larger than 1TB.
- FILE 64: Large File Emulator API Hook for Windows 98
- SYSDMCPL: Allows Windows 98SE to correctly report the friendly name of newer processors. (actually, friendly name seems less informative, so this should be user-selectable option)

GPT (needed above 2TB?) read/write support for Win9x from https://github.com/LordOfMice/Tools
- gpttsd - GPT (GUID Partition Table) TSD (Type-Specific Driver) for Win9x

**Comment by Torinde** [comment]
Will PATCHMEM function be optional? E.g. the rest of patcher9x to still work if there is no PSE support.

**Comment by TheMiningTeamYT** [comment]
> Related patches from http://lonecrusader.x10host.com/rloew/rloew.html
> 
> * PATCH1TB: Patches Windows 9x to correctly handle Partitions larger than 1TiB.
> * TBPLUS: Allows Windows 98SE to use hard drives larger than 1TB.
> * FILE 64: Large File Emulator API Hook for Windows 98
> * SYSDMCPL: Allows Windows 98SE to correctly report the friendly name of newer processors. (actually, friendly name seems less informative, so this should be user-selectable option)
> 
> GPT (needed above 2TB?) read/write support for Win9x from https://github.com/LordOfMice/Tools
> 
> * gpttsd - GPT (GUID Partition Table) TSD (Type-Specific Driver) for Win9x

I’d be worried about feature creep. Also, FILE64 has many incompatibilities in my experience.

**Comment by TheMiningTeamYT** [comment]
> OK, looks like there is a demand for this patch :-) I'll try to add it to some future release.

I see you finally added it! :D
However, I have a question: do you implement the low physical RAM patch? (the /m flag in patchmem)
Because at least I need that to be able to boot with my gigabit NIC.

**Comment by JHRobotics** [comment]
I added this option in the latest release (just run `patcher9x -m`).

**Comment by lss4** [comment]
Sorry but I'm not sure if the latest release contains patches for additional PATCHMEM arguments.

So far it seems `-m` corresponds to PATCHMEM's `/M`, but is the `/A` patch (Move AGP and Unfixed Tables >=128MB to Shared Arena) included in the latest patcher9x release? That patch might be needed for using certain AGP video cards on Win9x (might be helpful with PCIe ones as well).
